src/schema-association-service-1espt.ts

```diff
@@ -0,0 +1,90 @@
+                config.update('1ESPipelineTemplatesSchemaFile', undefined, vscode.ConfigurationTarget.WorkspaceFolder); // disable the 1ESPT schema configuration
+                vscode.window.showInformationMessage(Messages.disabled1ESPTSchemaAsADOOrgNotContains1ESPT)
+                return undefined; // 1ESPT repo not found
+            }
+            const repository = repositories.find(repo => repo.name === "1ESPipelineTemplates");
+            
+            // Using getItem from GitApi: getItem(repositoryId: string, path: string, project?: string, scopePath?: string, recursionLevel?: GitInterfaces.VersionControlRecursionType, includeContentMetadata?: boolean, latestProcessedChange?: boolean, download?: boolean, versionDescriptor?: GitInterfaces.GitVersionDescriptor, includeContent?: boolean, resolveLfs?: boolean, sanitize?: boolean): Promise<GitInterfaces.GitItem>;
+            var schemaFile = await gitApi.getItem(repository.id, "schema/1espt-base-schema.json", "1ESPipelineTemplates", "", VersionControlRecursionType.None, true, true, true, {}, true, true)
+
+            const schemaContent = schemaFile.content;
+            const schemaUri = Utils.joinPath(context.globalStorageUri, '1ESPTSchema', `${organizationName}-1espt-schema.json`);
+            await vscode.workspace.fs.writeFile(schemaUri, Buffer.from(schemaContent));
+            lastUpdated1ESPTSchema = new Date();
+            seen1ESPTOrganizations.add(organizationName);
+            return schemaUri;
+        }
+        else
+        // if user is signed in with account other than microsoft, then disable 1ESPT schema and delete the 1ESPT schema file
+        {
+            const config = vscode.workspace.getConfiguration('azure-pipelines')
+            config.update('1ESPipelineTemplatesSchemaFile', false, vscode.ConfigurationTarget.WorkspaceFolder);
+            await vscode.workspace.fs.delete(Utils.joinPath(context.globalStorageUri, '1ESPTSchema'), { recursive: true })
+            return undefined;
+        }
+    }
+    catch (error) {
+        logger.log(`Error : ${error} while fetching 1ESPT schema for org: ${organizationName}  : `, 'SchemaDetection');
+    }
+}
+
+export function getCached1ESPTSchemaInformation(context: vscode.ExtensionContext, organizationName: string, session: AzureSession, oneesptSchemaEnabled: boolean, lastUpdated1ESPTSchema: Date, seen1ESPTOrganizations: Set<string>): [URI, boolean] {
+    var skipOrgSpecificCachedSchema = false
+    var validSessionToEnable1ESPTSchema = false;
+    if (seen1ESPTOrganizations.has(organizationName)) {
+        const schemaUri1ESPT = Utils.joinPath(context.globalStorageUri, '1ESPTSchema', `${organizationName}-1espt-schema.json`);
+        validSessionToEnable1ESPTSchema = session.userId.endsWith("@microsoft.com") || session.userId.endsWith(".microsoft.com")
+
+        // check if 1ESPT schema file exists
+        var cached1ESPTSchemaFileExists = fs.existsSync(schemaUri1ESPT.path.substring(1));
```

Comment:
Prefer `vscode.workspace.fs.readFile` and using a try/catch to avoid usage of Node's built-in fs module.

Reply:
done